superset/connectors/sqla/models.py

```diff
@@ -1061,23 +1060,36 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
                     target_column_is_numeric=col_obj.is_numeric,
                     is_list_target=is_list_target,
                 )
-                if op in (
-                    utils.FilterOperator.IN.value,
-                    utils.FilterOperator.NOT_IN.value,
-                ):
-                    cond = col_obj.get_sqla_col().in_(eq)
-                    if isinstance(eq, str) and NULL_STRING in eq:
-                        cond = or_(
-                            cond,
-                            col_obj.get_sqla_col()  # pylint: disable=singleton-comparison
-                            == None,
+                if is_list_target:
```

Comment:
is it possible to add unit tests here?

Reply:
Done.

More granular logics could be handled by actual unit tests when we break down the query generator. I also added "revisit:add-tests" label to mark this as a followup item.